admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt

```diff
@@ -41,4 +43,28 @@ class ApiKeyBuilderTest {
         assertEquals("secret", apiKey.secret)
     }
 
+    @Test
+    fun `given api key with invalid format, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test") }
+    }
+
+    @Test
+    fun `given api key with the first part, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test_") }
+    }
+
+    @Test
+    fun `given api key with the last part, parse must throw NotAuthenticatedException`() {
```

Comment:
It's better to say what's missing:
`given api key without id part, ...`

Reply:
renamed